utils/errorHandler.ts

```diff
@@ -28,78 +33,80 @@ export const handleErrors = async (
 
 ) => {
 
-  try {
-    const { message } = await safeJsonParse(response);
-
-    switch (response.status) {
-      case 400:
-        if (message) {
-          setErrors(prevErrors => [...prevErrors, message]);
-        } else {
-          setErrors(['An unexpected error occurred. Please try again.']);
-        }
-      case 401:
-        if (message) {
-          logECS('error', message, {
-            url: { path: path }
-          });
-
-          try {
-            // Attempt to get new auth tokens
-            const response = await refreshAuthTokens();
-
-            if (response) {
-              router.push("/");
-            } else {
-              router.push('/login')
-            }
-          } catch (err) {
+  const { message } = await safeJsonParse(response);
+
+  switch (response.status) {
+    case 400:
+      if (message) {
+        setErrors(prevErrors => [...prevErrors, message]);
+      } else {
+        setErrors(['An unexpected error occurred. Please try again.']);
+      }
+      break;
+
+    case 401:
+      if (message) {
+        logECS('error', message, {
+          url: { path: path }
+        });
+
+        try {
+          // Attempt to get new auth tokens
+          const response = await refreshAuthTokens();
+
+          if (response) {
+            router.push("/");
+          } else {
             router.push('/login');
           }
+        } catch (err) {
+          if (err instanceof AuthError) {
+            // TODO: We want to push the errors, but also, we don't want the
+            // router to update if we are already on the login page.
+            // TODO: If NOT on the login page, redirect to it.
+            // router.push('/login');
```

Comment:
I just noticed my TODO's here. Slipped through the cracks somehow. I'll address this soon